src/miner.cpp

```diff
@@ -951,23 +951,23 @@ void ThreadStaker::operator()(std::vector<ThreadStaker::Args> args, CChainParams
                     status = staker.stake(chainparams, arg);
                 }
                 if (status == Staker::Status::error) {
-                    LogPrintf("ThreadStaker: (%s) terminated due to a staking error!\n", operatorName);
+                    LogPrint(BCLog::STAKING, "ThreadStaker: (%s) terminated due to a staking error!\n", operatorName);
                     it = args.erase(it);
                     continue;
                 }
                 else if (status == Staker::Status::minted) {
-                    LogPrintf("ThreadStaker: (%s) minted a block!\n", operatorName);
+                    LogPrint(BCLog::STAKING, "ThreadStaker: (%s) minted a block!\n", operatorName);
                     nMinted[arg.operatorID]++;
                 }
                 else if (status == Staker::Status::initWaiting) {
-                    LogPrintf("ThreadStaker: (%s) waiting init...\n", operatorName);
+                    LogPrint(BCLog::STAKING, "ThreadStaker: (%s) waiting init...\n", operatorName);
```

Comment:
How about instead, we keep the first 2 unchanged. However, for the `waiting init` throttle the log message so there's only one every few mins for `waiting init` if staking log isn't enabled.